modin/pandas/test/test_groupby.py

```diff
@@ -1717,3 +1717,29 @@ def test_not_str_by(by, as_index):
     eval_general(md_grp, pd_grp, lambda grp: grp.agg(lambda df: df.mean()))
     eval_general(md_grp, pd_grp, lambda grp: grp.dtypes)
     eval_general(md_grp, pd_grp, lambda grp: grp.first())
+
+
+@pytest.mark.parametrize("sort", [True, False])
+@pytest.mark.parametrize("is_categorical_by", [True, False])
+def test_groupby_sort(sort, is_categorical_by):
+    # from issue #3571
+    by = np.array(["a"] * 50000 + ["b"] * 10000 + ["c"] * 1000)
+    random_state = np.random.RandomState(seed=42)
+    random_state.shuffle(by)
```

Comment:
the bug is partitioning dependent, that's why I'm carrying the same data all around

Reply:
To clarify - you mean the shuffle is in place so that the bug gets tripped (bc if we didn't shuffle and the values weren't replicated across the partitions, we wouldn't hit the bug?)

Reply:
Right. For this data shuffled with this specific seed we know for sure that there is a problem caused by the incorrect order of categories in the reduce phase of groupby (see the issue explanation).

This problem might not be visible when the values are in a different order, so I'm using exact data from the #3571 report as a solid reproducer of the issue.